Devscord.DiscordFramework.UnitTests/Commands/RunnerOfIBotCommandMethodsTests/Tests/OptionalCommandsTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Devscord.DiscordFramework.UnitTests.Commands.RunnerOfIBotCommandMethodsTests.Controllers;
+using Devscord.DiscordFramework.UnitTests.Commands.RunnerOfIBotCommandMethodsTests.BotCommands;
+using Devscord.DiscordFramework.Commons.Exceptions;
+using Devscord.DiscordFramework.Framework.Commands.Parsing;
+using NUnit.Framework;
+
+namespace Devscord.DiscordFramework.UnitTests.Commands.RunnerOfIBotCommandMethodsTests.Tests
+{
+    [TestFixture]
+    class OptionalCommandsTests
+    {
+        private readonly GetterOfThingsAboutBotCommand _getterOfThings;
+        private readonly TestController _controller;
+        private readonly CommandParser _commandParser;
+        private static readonly object[] _messagesAndExpectedTimesOrLists = new[]
+        {
+            new object[] { "-OptionalArgs -testtime 5d", TimeSpan.FromDays(5) },
+            new object[] { "-custom_optional time 17m", TimeSpan.FromMinutes(17) },
+            new object[] { "-OptionalArgs -testlist sth", new List<string> { "sth" } },
+            new object[] { "-custom_optional list abc qwerty", new List<string> { "abc", "qwerty" } }
+        };
+
+        public OptionalCommandsTests()
+        {
+            this._getterOfThings = new GetterOfThingsAboutBotCommand();
+            this._controller = new TestController();
+            this._commandParser = new CommandParser();
+        }
+
+        [Test]
+        [TestCase("-OptionalArgs -testnullableint 5", 5)]
+        [TestCase("-custom_optional int -2", -2)]
+        [TestCase("-OptionalArgs -testusermention <@12345>", 12345UL)]
+        [TestCase("-custom_optional user <@!789>", 789UL)]
+        [TestCase("-OptionalArgs -testchannelmention <#333>", 333UL)]
+        [TestCase("-custom_optional channel <#877>", 877UL)]
+        [TestCaseSource(nameof(_messagesAndExpectedTimesOrLists))]
+        public async Task ShouldGiveExpectedValueForOptionalArg(string message, object expectedValue)
+        {
+            // Arrange:
+            var customCommands = this._getterOfThings.CreateCustomCommands(new (Type type, string regexInText)[] 
```

Comment:
czemu nie zrobisz tego liste + ustawienie w mocku?

Reply:
no cóż mogę tę tablicę zamienić na listę, ale to i tak nic wiele nie zmieni. 
Nie wiem, o co ci chodzi z "ustawieniem w mocku", przecież to dalej jest ustawiane w mocku.